client/src/components/TagField.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component, PropTypes } from 'react';
+import Select from 'react-select';
+import fetch from 'isomorphic-fetch';
+import url from 'url';
+
+class TagField extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: props.value,
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.getOptions = this.getOptions.bind(this);
+  }
+
+  onChange(value) {
+    this.setState({
+      value
+    });
+
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange(value);
+    }
+  }
+
+  getOptions(input) {
+    if (!this.props.lazyLoad) {
+      return Promise.resolve({ options: this.props.options });
+    }
+
+    if (!input) {
+      return Promise.resolve({ options: [] });
+    }
+
+    const fetchURL = url.parse(this.props.optionUrl, true);
```

Comment:
Could you please dereference these props - lazyLoad, options and optionUrl? There's also some places in the render method that could use this too